lighthouse-core/test/gather/mock-commands.js

```diff
@@ -71,15 +117,25 @@ function createMockOnceFn() {
     setTimeout(() => listener(response), 0);
   });
 
-  mockFn.mockEvent = (event, response) => {
-    mockEvents.push({event, response});
-    return mockFn;
-  };
-
-  mockFn.findListener = event => {
-    expect(mockFn).toHaveBeenCalledWith(event, expect.anything());
-    return mockFn.mock.calls.find(call => call[0] === event)[1];
-  };
+  const mockFn = Object.assign(mockFnImpl, {
+    ...mockFnImpl,
```

Comment:
why are we assigning its spread to itself?

Reply:
how strange